src/lib.rs

```diff
@@ -49,6 +49,9 @@ pub fn compile_book(
   if opts.eta {
     core_book.values_mut().for_each(Net::eta_reduce);
   }
+  if opts.inline {
+    core_book.inline().map_err(|e| format!("Inline:\n{:ERR_INDENT_SIZE$}{e}", ""))?;
```

Comment:
Could you instead add this to the diagnostics so that we also print the warnings from desugaring?
Something like 
```
diagnostics.start_pass();
let res = core_book.inline();
diagnostics.take_book_err(res);
diagnostics.fatal()?;
```